alerter.c

```diff
@@ -1,32 +1,19 @@
 #include <stdio.h>
 #include <assert.h>
+#include "alerter.h"
 
 int alertFailureCount = 0;
-
-int networkAlertStub(float celcius) {
-    printf("ALERT: Temperature is %.1f celcius.\n", celcius);
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
-}
-
-void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
-    if (returnCode != 200) {
-        // non-ok response is not an error! Issues happen in life!
-        // let us keep a count of failures to report
-        // However, this code doesn't count failures!
-        // Add a test below to catch this bug. Alter the stub above, if needed.
-        alertFailureCount += 0;
-    }
-}
+float farenheit;
+float celcius;
 
 int main() {
-    alertInCelcius(400.5);
-    alertInCelcius(303.6);
+    
+    alertFailureCount = AlertFarenheitToCelcius(400.5,alertFailureCount,networkAlertInterface);
+    alertFailureCount = AlertFarenheitToCelcius(303.6,alertFailureCount,networkAlertStub);
+    alertFailureCount = AlertFarenheitToCelcius(20,alertFailureCount,networkAlertInterface);
+    alertFailureCount = AlertFarenheitToCelcius(10,alertFailureCount,networkAlertStub);
     printf("%d alerts failed.\n", alertFailureCount);
+    assert(alertFailureCount > 0);

```

Comment:
Checking >0 is a weak criteria. The production code outcome should be deterministic and that should be expressed in the test cases. Based on the inputs the production output should be fixed. the exact failure count should be asserted.